pkg/server/endpoints/node/handler_test.go

```diff
@@ -605,13 +607,55 @@ func (s *HandlerSuite) TestFetchX509SVIDWithWorkloadCSR() {
 		Csrs: s.makeCSRs(workloadID),
 	})
 
-	// Downstream responses don't contain the downstream registration entry
-	// since downstream entries aren't intended for workloads.
 	s.Equal([]*common.RegistrationEntry{entry}, upd.RegistrationEntries)
 	s.assertBundlesInUpdate(upd)
 	s.assertSVIDsInUpdate(upd, workloadID)
 }
 
+func (s *HandlerSuite) TestFetchX509SVIDWithSingleDNS() {
+	dnsList := []string{"somehost1"}
+
+	s.attestAgent()
+
+	entry := s.createRegistrationEntry(&common.RegistrationEntry{
+		ParentId: agentID,
+		SpiffeId: workloadID,
+		DnsList:  dnsList,
+	})
+
+	upd := s.requireFetchX509SVIDSuccess(&node.FetchX509SVIDRequest{
+		Csrs: s.makeCSRs(workloadID),
+	})
+
+	s.Equal([]*common.RegistrationEntry{entry}, upd.RegistrationEntries)
+	s.assertBundlesInUpdate(upd)
+	chains := s.assertSVIDsInUpdate(upd, workloadID)
+	s.Equal(dnsList, chains[0][0].DNSNames)
+	s.Equal("somehost1", chains[0][0].Subject.CommonName)
+}
+
+func (s *HandlerSuite) TestFetchX509SVIDWithMultipleDNS() {
```

Comment:
might be good to test the CA bits too to ensure that DNS and CN is not being set?

Reply:
adding to TestFetchX509SVIDWithDownstreamCSR